### PR TITLE
boards/e180-zg120b-tb: add support

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description
This PR adds support for the [Ebyte E180-ZG120B-TB](http://www.ebyte.com/en/product-view-news.aspx?id=896). It's a cheap board that houses a EFM32MG1B SoC with radio.

The board is very similar to the IKEA TRADFRI modules and the SLTB001A. It uses the EFM32 drivers. Mapped peripherals is limited to UART only, since the board does not 'expose' any I2C and/or SPI pins. This board features are limited, but the Radio makes it interested (but depends on #8569).

I have noticed that the board datasheet and SoC datasheet are wrong. Some of the LEDs on the board aren't even connected, and they certainly don't line up with the actual EFM32 chip. I verified this by de-lidding the SoC and probe with a multimeter.

The board has a serial-to-usb converter. It does not have a programming interface, but I just used JLink to connect it (these pins are correct on the board's silk screen).

### Testing procedure
Run CI tests on the board.

image

### Issues/PRs references
None

### CODE REVIEW

File: boards/e180-zg120b-tb/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# set default port depending on operating system
+PORT_LINUX ?= /dev/ttyACM0
```
Comment (benpicco) [inline boards/e180-zg120b-tb/Makefile.include:2]: for me this is 
```suggestion
PORT_LINUX ?= /dev/ttyUSB0
```

File: boards/e180-zg120b-tb/doc.txt [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+ * @ingroup     boards
+ * @brief       Support for Ebyte E180-ZG120B-TB Test Board
+
+## Overview
+Ebyte E180-ZG120B Test Board is equipped with the EFM32 microcontroller.
+It is specifically designed for low-power applications, having energy-saving
+peripherals, different energy modes and short wake-up times.
+
+The starter kit is equipped with an Advanced Energy Monitor. This allows you to
+actively measure the power consumption of your hardware and code, in real-time.
```
Comment (benpicco) [inline boards/e180-zg120b-tb/doc.txt:12]: I wouldn't have expected that from this cheap board

File: boards/e180-zg120b-tb/doc.txt [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+## Hardware
+
+### MCU
+| MCU           | EFR32MG1B232F256GM32                                                                    |
+|---------------|-----------------------------------------------------------------------------------------|
+| Family        | ARM Cortex-M4F                                                                          |
+| Vendor        | Ebyte                                                                                   |
+| Vendor Family | EFM32 Mighty Gecko 1B                                                                   |
+| RAM           | 31.0 KiB                                                                                |
```
Comment (benpicco) [inline boards/e180-zg120b-tb/doc.txt:22]: that's an odd number - not 32 k?

File: boards/e180-zg120b-tb/doc.txt [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+| Timer      | 0       | TIMER0 + TIMER1 |                             | TIMER0 is used as prescaler (must be adjecent)      |
+|            | 1       | LETIMER0        |                             |                                                     |
+| UART       | 0       | USART0          | RX: PA1, TX: PA0            | Default STDIO output                                |
+
+### User interface
+| Peripheral | Mapped to | Pin  | Comments        |
+|------------|-----------|------|-----------------|
+| Button     | PB0_PIN   | PD15 | Mode Change     |
+|            | PB1_PIN   | PD13 | Touch Link      |
+|            | PB2_PIN   | PB11 | Baud Rate Reset |
```
Comment (benpicco) [inline boards/e180-zg120b-tb/doc.txt:55]: So the fourth button (with the Chinese characters) is not connected at all?

File: boards/e180-zg120b-tb/Kconfig [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    bool
+    default y
+    select BOARD_COMMON_SILABS
+    select CPU_MODEL_EFR32MG1B232F256GM32
+    select HAS_PERIPH_ADC
+    select HAS_PERIPH_RTC
+    select HAS_PERIPH_RTT
+    select HAS_PERIPH_TIMER
+    select HAS_PERIPH_UART
+    select HAS_PERIPH_UART_MODECFG
```
Comment (benpicco) [inline boards/e180-zg120b-tb/Kconfig:20]: `HAS_PERIPH_UART_MODECFG` should be a CPU feature

**Review by benpicco — COMMENTED** [review]
Thank you, works like a charm!
I can now confirm that it's the shoddy socket of my debug adapter that makes flashing those boards a pain (had similar issues as with the nRF52 boards from ebyte), but once I managed to flash it (wiggling the cables in the socket helped) I could flash riotboot with #15493 and flashing via riotboot works well too.

I hope we can use this to push forward the radio driver.

File: boards/e180-zg120b-tb/doc.txt (hunk 2 above)
Comment (basilfx) [reply boards/e180-zg120b-tb/doc.txt:12]: Whoops, that's from a Silicon labs board, I will remove that!

Apart from that, you should be able to use a Silicon Labs board for power measurements ;-)

File: boards/e180-zg120b-tb/doc.txt (hunk 3 above)
Comment (basilfx) [reply boards/e180-zg120b-tb/doc.txt:22]: Same as with the SLTB001a, 1 KiB is reserved for the radio.

File: boards/e180-zg120b-tb/doc.txt (hunk 4 above)
Comment (basilfx) [reply boards/e180-zg120b-tb/doc.txt:55]: That's the reset button. I used Google Lens to identify it :-)

Comment (benpicco) [reply boards/e180-zg120b-tb/doc.txt:55]: Probably worth adding a note that the fourth button is reset, wouldn't have guessed it

File: boards/e180-zg120b-tb/doc.txt (hunk 3 above)
Comment (benpicco) [reply boards/e180-zg120b-tb/doc.txt:22]: even if the radio is not used?

Comment (basilfx) [reply boards/e180-zg120b-tb/doc.txt:22]: Then you'll have the 32 KiB available. The Radio blob uses 1 KiB. I'll clarify it.

**Comment by benpicco** [comment]
Unrelated to this board,  but is it normal that the RTC resets on `reboot` on efm32?

**Comment by basilfx** [comment]
> Unrelated to this board, but is it normal that the RTC resets on `reboot` on efm32?

Never thought about that, but yes: https://www.silabs.com/documents/public/application-notes/AN0005-EFM32-RTC.pdf

File: boards/e180-zg120b-tb/Kconfig (hunk 5 above)
Comment (basilfx) [reply boards/e180-zg120b-tb/Kconfig:20]: If added to the CPU features, doesn't that automatically add a dependency on the UART peripheral?

File: boards/e180-zg120b-tb/doc.txt [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+## Hardware
+
+### MCU
+| MCU           | EFR32MG1B232F256GM32                                                                    |
+|---------------|-----------------------------------------------------------------------------------------|
+| Family        | ARM Cortex-M4F                                                                          |
+| Vendor        | Ebyte                                                                                   |
+| Vendor Family | EFM32 Mighty Gecko 1B                                                                   |
+| RAM             | 32.0 KiB (1.0 KiB reserved by radio blob)                                             |
```
Comment (benpicco) [inline boards/e180-zg120b-tb/doc.txt:19]:
```suggestion
| RAM           | 32.0 KiB (1.0 KiB reserved by radio blob)                                             |
```

**Comment by benpicco** [comment]
Please squash.

File: boards/e180-zg120b-tb/Kconfig (hunk 5 above)
Comment (benpicco) [reply boards/e180-zg120b-tb/Kconfig:20]: No, it just means that the CPU implements this feature.
This way we can extend the API without having to implement it for all platforms at once.

**Review by benpicco — APPROVED** [review]

**Comment by basilfx** [comment]
Thanks!

**Pull request merged**